app/src/main/java/comp3350/mealbuddy/objects/Account.java

```diff
@@ -31,6 +31,24 @@ public Account(UserInfo user) {
         goals = new ArrayList<>();
     }
 
+    /*
+     * Copy Constructor
+     * Initializes the values for the account, from an existing account
+     * Parameters:
+     *     @param original - the account to copy.
+     */
+    public Account(Account original) {
+        this.user = new UserInfo(original.user);
+        goals = new ArrayList<>();
+        for (Goal g : original.goals) {
+            this.goals.add(Goal.copyGoal(g));
```

Comment:
Make a test for copy goals. Same for copy edible. I did not see a test for either.